src/main/java/org/openpnp/spi/base/AbstractPnpJobProcessor.java

```diff
@@ -84,8 +113,9 @@ public static NozzleTip findNozzleTip(Head head, Part part) throws Exception {
      * @throws Exception If no Feeder is found that is both enabled and is serving the Part.
      */
     public static Feeder findFeeder(Machine machine, Part part) throws Exception {
+		if(part!=null)
         for (Feeder feeder : machine.getFeeders()) {
-            if (feeder.getPart() == part && feeder.isEnabled()) {
+            if (feeder.getPart().getId().equals(part.getId()) && feeder.isEnabled()) {
```

Comment:
What is the purpose of this change? There should never be more than one instance of Part with the same ID in the configuration, so this should not be needed.